src/main/kotlin/no/nav/syfo/utsattoppgave/UtsattOppgaveService.kt

```diff
@@ -39,84 +72,69 @@ class UtsattOppgaveService(
         logger.info("Fant oppgave for inntektsmelding: ${oppgave.arkivreferanse} med tilstand: ${Tilstand.Forkastet.name}")
         val gjelderSpeil = oppdatering.oppdateringstype == OppdateringstypeDTO.OpprettSpeilRelatert
 
-        if (oppgave.tilstand == Tilstand.Utsatt && oppdatering.handling == no.nav.syfo.utsattoppgave.Handling.Utsett) {
-            if (oppgave.timeout == null) {
-                metrikk.tellUtsattOppgave_UtenDato()
+        return when {
+            oppgave.tilstand == Tilstand.Utsatt && oppdatering.handling == Handling.Utsett -> {
+                oppdatering.timeout ?: error("Timeout på utsettelse mangler for inntektsmelding: ${oppgave.arkivreferanse}")
+
+                oppgave
+                    .apply {
+                        timeout = oppdatering.timeout
+                        oppdatert = LocalDateTime.now()
+                        speil = gjelderSpeil
+                    }
+                    .lagre()
+
+                metrikk.tellUtsattOppgave_Utsett()
+                logger.info("Oppdaterte timeout på inntektsmelding: ${oppgave.arkivreferanse} til ${oppdatering.timeout}")
             }
-            oppdatering.timeout ?: error("Timeout på utsettelse mangler for inntektsmelding: ${oppgave.arkivreferanse}")
-            oppgave.timeout = oppdatering.timeout
-            oppgave.oppdatert = LocalDateTime.now()
-            oppgave.speil = gjelderSpeil
-            lagre(oppgave)
-            metrikk.tellUtsattOppgave_Utsett()
-            logger.info("Oppdaterte timeout på inntektsmelding: ${oppgave.arkivreferanse} til ${oppdatering.timeout}")
-            return
-        }
 
-        if (oppgave.tilstand == Tilstand.Utsatt && oppdatering.handling == no.nav.syfo.utsattoppgave.Handling.Forkast) {
-            oppgave.oppdatert = LocalDateTime.now()
-            lagre(oppgave.copy(tilstand = Tilstand.Forkastet, speil = gjelderSpeil))
-            metrikk.tellUtsattOppgave_Forkast()
-            logger.info("Endret oppgave: ${oppgave.arkivreferanse} til tilstand: ${Tilstand.Forkastet.name}")
-            return
-        }
+            oppgave.tilstand == Tilstand.Utsatt && oppdatering.handling == Handling.Forkast -> {
+                oppgave
+                    .apply {
+                        oppdatert = LocalDateTime.now()
+                    }
+                    .copy(
+                        tilstand = Tilstand.Forkastet,
+                        speil = gjelderSpeil,
+                    )
+                    .lagre()
```

Comment:
Jeg forstår ikke helt hvorfor `oppgave` blir muterer `oppdatert`-feltet, for så å bruke den ikke-muterende `copy`-funksjonen 🤷